interpreter/astwalker.go

```diff
@@ -119,27 +119,23 @@ func (w *astWalker) walk(node *expr.Expr) []Instruction {
 }
 
 func (w *astWalker) walkConst(node *expr.Expr) Instruction {
-	constExpr := node.GetConstExpr()
+	literalExpr := node.GetLiteralExpr()
 	var value ref.Value = nil
-	switch constExpr.ConstantKind.(type) {
-	case *expr.Constant_BoolValue:
-		value = types.Bool(constExpr.GetBoolValue())
-	case *expr.Constant_BytesValue:
-		value = types.Bytes(constExpr.GetBytesValue())
-	case *expr.Constant_DoubleValue:
-		value = types.Double(constExpr.GetDoubleValue())
-	case *expr.Constant_DurationValue:
-		value = types.Duration{Duration: constExpr.GetDurationValue()}
-	case *expr.Constant_Int64Value:
-		value = types.Int(constExpr.GetInt64Value())
-	case *expr.Constant_NullValue:
-		value = types.Null(constExpr.GetNullValue())
-	case *expr.Constant_StringValue:
-		value = types.String(constExpr.GetStringValue())
-	case *expr.Constant_TimestampValue:
-		value = types.Timestamp{Timestamp: constExpr.GetTimestampValue()}
-	case *expr.Constant_Uint64Value:
-		value = types.Uint(constExpr.GetUint64Value())
+	switch literalExpr.LiteralKind.(type) {
+	case *expr.Literal_BoolValue:
+		value = types.Bool(literalExpr.GetBoolValue())
+	case *expr.Literal_BytesValue:
+		value = types.Bytes(literalExpr.GetBytesValue())
+	case *expr.Literal_DoubleValue:
+		value = types.Double(literalExpr.GetDoubleValue())
+	case *expr.Literal_Int64Value:
+		value = types.Int(literalExpr.GetInt64Value())
+	case *expr.Literal_NullValue:
+		value = types.Null(literalExpr.GetNullValue())
+	case *expr.Literal_StringValue:
+		value = types.String(literalExpr.GetStringValue())
+	case *expr.Literal_Uint64Value:
+		value = types.Uint(literalExpr.GetUint64Value())
 	}
 	return NewConst(node.Id, value)
```

Comment:
NewConst -> NewLiteral ?  I can't see the context to tell whether this is appropriate.